src/components/response/ResponseBlock.tsx

```diff
@@ -31,29 +31,23 @@ export default function ResponseBlock({
   const currentStep = useCurrentStep();
   const storedAnswer = status?.answer;
 
-  const configInUse = (config) as IndividualComponent;
+  const configInUse = config as IndividualComponent;
 
   const responses = configInUse?.response?.filter((r) =>
     r.location ? r.location === location : location === 'belowStimulus'
   ) || [];
 
   const storeDispatch = useStoreDispatch();
-  const { saveTrialAnswer, updateResponseBlockValidation, setIframeAnswers } = useStoreActions();
+  const { updateResponseBlockValidation } = useStoreActions();
   const answerValidator = useAnswerField(responses, currentStep, storedAnswer || {});
-  const [disableNext, setDisableNext] = useInputState(!storedAnswer);
   const [checkClicked, setCheckClicked] = useState(false);
-  const { iframeAnswers, trialValidation } = useStoreSelector((state) => state);
-  const areResponsesValid = useAreResponsesValid(currentStep);
-  const { storageEngine } = useStorageEngine();
+  const { iframeAnswers } = useStoreSelector((state) => state);
   const hasCorrectAnswer = ((configInUse?.correctAnswer?.length || 0) > 0);
 
-  const startTime = useMemo(() => {
-    return Date.now();
-  }, []);
-
   const showNextBtn =
     location === (configInUse?.nextButtonLocation || 'belowStimulus');
 
+  // TODO: shift following effects to useNextStep hook, fix dependencies
```

Comment:
TODO?

Reply:
Yeah, I am unsure if this should be shifted to useNextStep hook as well or not.

Reply:
@JackWilb , I will make a separate PR for this. I think all of the effects should be shifted to useNextStep hook, which will also return user's answers and correct answer if available.

This makes it easier feasible to completely remove ResponseBlock if study designer wants to create their own within the stimuli.